modules/docs/all.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (config.docs._utils)
+    docsPageLiberalType
+    ;
+in
+{
+  options.docs = {
+    _allInputs = lib.mkOption {
+      type = with lib.types; listOf str;
+      description = "`docs.*` option names that should be included in `docs.all`.";
+      defaultText = config.docs._allInputs;
+      default = [ ];
+      internal = true;
+      visible = false;
+    };
+    all = lib.mkOption {
+      type = with lib.types; listOf docsPageLiberalType;
+      description = ''
+        All enabled doc pages defined in:
+        ${lib.concatMapStringsSep "\n" (name: "- `docs.${name}`") config.docs._allInputs}.
+      '';
+      visible = "shallow";
+      readOnly = true;
+    };
```

Comment:
This dance is to have a readonly option whose definitions are split around the nixvim source right? Is the `readonly` really important? If it were absent this could be simplified a bit with an `apply` right?

Reply:
Kinda.

I marked it as read-only because it seemed unlikely that someone should define a page directly here, without using one of the more specific implementations.

The other options listed in `_allInputs` all provide some additional functionality, built on top of the base `docsPageModule`.

Also, unlike `all`, they don't set `_module.check = false`, so their config definitions must strictly match their respective option declarations.

Reply:
I've pushed up some (half-finished) work to simplify and refactor this.

My new idea is:
- One main canonical `docs.pages` option
  - Effectively a "files" option; with `target`, `text`, and `source` sub-options
  - This means no more `all` option, no more duplicated documentation, etc
- Specialized options; `docs.optionsPages`, `docs.platformPages`, etc
  - Each has a `page` sub-option; a deferred module, assigned later to `docs.pages.<name>`
  - These options no longer need to extend from the pages module, instead they can define things in their deferred `page` module.
  - Each specialized option is responsible for doing `docs.pages = mapAttrs (_: cfg: cfg.page) config.docs.specialOption`

(Deferred modules is how home-manager's per-user modules are implemented, for example)